pymesos/process.py

```diff
@@ -147,7 +147,12 @@ def abort(self):
 
     def stop(self):
         Process.abort(self)
-        self.join()
+        assert (threading.current_thread() != self.accept_t), "Can't call stop in accept_t thread."
+        logger.debug("join ioloop thread.")
+        self.accept_t.join()
+        if threading.current_thread() != self.delay_t:
```

Comment:
u know, C++ have destructors, so native mesos scheduler can do `join` there without worrying that users forgot to call `join`. unfortunately we can’t. as `delay_t` is started and owned by `pymeos`, there is no doubt `pymesos` is responsible for the termination of this thread.